src/intelligence_layer/connectors/data/models.py

```diff
@@ -4,16 +4,21 @@
 from typing import Any, Optional
 
 from pydantic import BaseModel, ConfigDict
-from pydantic.alias_generators import to_camel, to_snake
+from pydantic.alias_generators import to_camel
 
 
 class BaseDataModel(BaseModel):
-    model_config = ConfigDict(alias_generator=to_camel, arbitrary_types_allowed=True)
+    model_config = ConfigDict(
+        alias_generator=to_camel,
+        arbitrary_types_allowed=True,
+        populate_by_name=True,
+    )
 
 
 class MediaTypes(str, Enum):
     jsonlines = "application/x-ndjson"
     jsonlines_ = "application/jsonlines"
+    jsonlines__ = "jsonlines"
```

Comment:
Out of curiosity:
What does this do and why are we using underscores to distinguish the fields of the class?

Reply:
@SebastianNiehusAA I don't know why this was like this. I've just added another entry in the class. As far as I understand, this is supposed to be an enum to store the supported normalization for jsonlines. I believe that this normalization can be better achieved using a dictionary instead of a class. If you think it makes sense, I'd like to change it in this PR.

Reply:
@mveleci I think you introduced this notation here. Do you have any insights on this?

Reply:
@silver472 I like your idea of using a dict instead as this would solve the - in my eyes - strange naming

Reply:
@SebastianNiehusAA I've decided to use an Annotated validator instead of an Enum. Could you check again?

Reply:
Much nicer :)